contracts/internal/EverlongPositions.sol

```diff
@@ -0,0 +1,295 @@
+    /// @dev Can be overridden by child contracts.
+    /// @param _amount Amount of capital provided for `openLong`.
+    /// @return Minimum number of bonds to receive from `openLong`.
+    function _minOpenLongOutput(
+        uint256 _amount
+    ) internal view virtual returns (uint256) {
+        return 0;
+    }
+
+    // TODO: Come up with a safer value.
+    /// @dev Calculates the minimum vault share price at which to
+    ///      open the long.
+    /// @dev Can be overridden by child contracts.
+    /// @param _amount Amount of capital provided for `openLong`.
+    /// @return minimum vault share price for `openLong`.
+    function _minVaultSharePrice(
+        uint256 _amount
+    ) internal view virtual returns (uint256) {
+        return 0;
+    }
+
+    // TODO: Come up with a safer value.
+    /// @dev Calculates the minimum proceeds Everlong will accept for
+    ///      closing the long.
+    /// @dev Can be overridden by child contracts.
+    /// @param _maturityTime Maturity time of the long to close.
+    /// @param _bondAmount Amount of bonds to close.
+    function _minCloseLongOutput(
+        uint256 _maturityTime,
+        uint256 _bondAmount
+    ) internal view returns (uint256) {
+        return 0;
+    }
+
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Position Opening (Internal)                             │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @dev Spend the excess idle liquidity for the Everlong contract.
+    /// @dev Can be overridden by implementing contracts to configure
+    ///      how much idle to spend and how it is spent.
+    function _spendExcessLiquidity() internal {
+        // Open the long position with the available excess liquidity.
+        // TODO: Worry about slippage.
+        // TODO: Ensure amount < maxLongAmount
+        uint256 _amount = _excessLiquidity();
+        (uint256 _maturityTime, uint256 _bondAmount) = IHyperdrive(_hyperdrive)
+            .openLong(
+                _amount,
+                _minOpenLongOutput(_amount),
+                _minVaultSharePrice(_amount),
+                IHyperdrive.Options(address(this), _asBase, "")
+            );
+
+        // Update positions to reflect the newly opened long.
+        _handleOpenLong(uint128(_maturityTime), uint128(_bondAmount));
+    }
+
+    /// @dev Account for newly purchased bonds within the `PositionManager`.
+    /// @param _maturityTime Maturity time for the newly purchased bonds.
+    /// @param _bondAmountPurchased Amount of bonds purchased.
+    function _handleOpenLong(
+        uint128 _maturityTime,
+        uint128 _bondAmountPurchased
+    ) internal {
+        // Compare the maturity time of the purchased bonds
+        // to the most recent position's `maturityTime`.
+        if (
+            _positions.length() != 0 &&
+            _decodePosition(_positions.back()).maturityTime > _maturityTime
```

Comment:
Just wanted to be overly cautious / exhaustive, can't really think of a case where this could occur in practice unless something else was horribly wrong (changing Hyperdrive instances etc.)

Reply:
Sounds good. I'm fine with keeping it. It's definitely safer.